packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -99,6 +101,18 @@ contract MixinExchangeCore is
         }
         require(takerTokenFillAmount > 0);
 
+        // Validate maker asset proxy
+        if (!isValidAssetProxyId(order.makerAssetProxyId)) {
```

Comment:
These checks should be made in `AssetTransferProxy` using the mapping rather than the enum (I'd remove the enum altogether). The goal is to be able to add new proxies _without_ swapping out the Exchange contract.